site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+    options: [
+        {label: 'Sensitive', value: 'sensitive'},
+        {label: 'Urgent', value: 'urgent'},
+        {label: 'Feature Request', value: 'feature_request'},
+    ],
+};
+```
+
+We can then process the selected values in the form submission like so:
+
+```ts
+type CreateTicketFormSubmission = {
+    summary: string;
+    description: string;
+    tags: AppSelectOption[]; // {label: string; value: string}[]
+};
+
+app.post('/forms/create-ticket/submit', async (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the submitted form values
+    const {summary, description, tags}: CreateTicketFormSubmission = call.values;
+
+    // Extract selected values from the tags array
+    const tagValues = tags.map(option => option.value);
+
+    // Process request differently based on tags selected
+    const isUrgent = tagValues.includes('urgent');
+    if (isUrgent) {
+        // ...
+    }
+
+    // ... Process the submission as usual
+});
+```
+
+### Autocomplete select values
+
+> Note: If we need autocomplete functionality for users or channels, we can set the field's `type` to "user" or "channel", and the from will show an autocomplete selector for the respective field type. The Mattermost server will take care of the autocomplete requests for users and channels. WHen the form is submitted, the field's value will be submitted as an [AppSelectOption](https://github.com/mattermost/mattermost-redux/blob/3d1028034d7677adfda58e91b9a5dcaf1bc0ff99/src/types/apps.ts#L138), where the selected option's `value` is the `id` for the `UserProfile` or `Channel`.
+
+What if the list of available tags is too large to be computed at once, and we need to use an autocomplete API to retrieve these values dynamically? We will use a `dynamic_select` field for this.
+
+Let's change the field's `type` to `dynamic_select`, and remove the `options` property:
+
+```ts
+const tagsField: AppField = {
+    name: 'tags',
+    modal_label: 'Tags',
+    type: 'dynamic_select',
+    value: [],
+    multiselect: true,
+};
+```
+
+This will use the form's `call` to fetch these values at the endpoint `/forms/create-ticket/lookup`. The same `lookup` path is used for any `dynamic_select` field of this form. We know what field is requesting the autocomplete by accessing `call.selected_field`. We can get the user's query typed in by accessing `call.query`:
+
+```ts
+app.post('/forms/create-ticket/lookup', async (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // For this form, we only have one dynamic select field, so we should only be receiving requests for that field. This block is here to show that this endpoint could be called for another field, if the form had multiple dynamic select fields, and we would need to handle them individually here when that field is being queried for autocomplete.
```

Comment:
If you want to convey this message, wouldn't it make more sense to make this a switch like:
```
switch (call.selected_field) {
case 'tags':
    handleTagsLookup(req, res)
default:
    //handle error
}
```